examples/security/using_session_auth.py

```diff
@@ -0,0 +1,138 @@
+from typing import Any, Literal, Optional
+from uuid import UUID, uuid4
+
+from pydantic import BaseModel, EmailStr, SecretStr
+
+from starlite import (
+    ASGIConnection,
+    NotAuthorizedException,
+    OpenAPIConfig,
+    Request,
+    SessionAuth,
+    Starlite,
+    get,
+    post,
+)
+from starlite.middleware.session.memory_backend import MemoryBackendConfig
+
+
+# Let's assume we have a User model that is a pydantic model.
+# This though is not required - we need some sort of user class -
+# but it can be any arbitrary value, e.g. an SQLAlchemy model,
+# a representation of a MongoDB  etc.
+class User(BaseModel):
+    id: UUID
+    name: str
+    email: EmailStr
+
+
+# we also have pydantic types for two different
+# kinds of POST request bodies: one for creating
+# a user, e.g. "sign-up", and the other for logging
+# an existing user in.
+class UserCreatePayload(BaseModel):
+    name: str
+    email: EmailStr
+    password: SecretStr
+
+
+class UserLoginPayload(BaseModel):
+    email: EmailStr
+    password: SecretStr
+
+
+# The SessionAuth class requires a handler callable
+# that takes the session dictionary, and returns the
+# 'User' instance correlating to it.
+#
+# The session dictionary itself is a value the user decides
+# upon. So for example, it might be a simple dictionary
+# that holds a user id, for example: { "id": "abcd123" }
+#
+# Note: The callable can be either sync or async - both will work.
+async def retrieve_user_handler(session: dict[str, Any], connection: ASGIConnection) -> Optional[User]:
```

Comment:
```suggestion
async def retrieve_user_handler(session: Dict[str, Any], connection: ASGIConnection) -> Optional[User]:
```
Examples should work on Python 3.8